rtlib/generic_opts.hh

```diff
@@ -183,6 +287,69 @@ class Opts {
           case 'r' :
             repeats = std::atoi(optarg);
             break;
+#ifdef CHECKPOINTING_INTEGRATED
+          case 'p' :
+            checkpoint_interval = parse_checkpointing_interval(optarg);
+            break;
+          case 'I' :
+          {
+            boost::filesystem::path arg_path(optarg);
+            if (arg_path.is_absolute()) {
+              checkpoint_in_path = arg_path;
+            } else {
+              checkpoint_in_path = boost::filesystem::current_path() / arg_path;
+            }
+            if (!boost::filesystem::exists(checkpoint_in_path) ||
+                !boost::filesystem::is_regular_file(checkpoint_in_path)) {
+              throw OptException("Logfile could not be found at path \"" +
+                                 checkpoint_in_path.string() +
+                                 "\"!");
+            }
+
+            // check user permissions of checkpoint input directory
+            struct stat checkpoint_in_dir;
+            stat(checkpoint_in_path.c_str(), &checkpoint_in_dir);
+            if (!(checkpoint_in_dir.st_mode & S_IRUSR)) {
+              throw OptException("Missing read permissions for"
+                                 " Logfile \""
+                                 + checkpoint_in_path.string()
+                                 + "\"!");
+            }
+            break;
+          }
+          case 'O' :
+          {
+              boost::filesystem::path arg_path(optarg);
+              boost::filesystem::path out_path = arg_path.parent_path();
+
+              if (out_path.is_absolute()) {
+                checkpoint_out_path = out_path;
+              } else {
+                checkpoint_out_path /= out_path;
+              }
+
+            user_file_prefix = arg_path.filename().string();
+
+            if (!boost::filesystem::exists(checkpoint_out_path) ||
+                !boost::filesystem::is_directory(checkpoint_out_path)) {
+              throw OptException("The output path \"" +
+                                 checkpoint_out_path.string() +
+                                 "\" is not a directory!");
+            }
+
+            // check user permissions of checkpoint output directory
+            struct stat checkpoint_out_dir;
+            stat(checkpoint_out_path.c_str(), &checkpoint_out_dir);
+
+            if (!(checkpoint_out_dir.st_mode & S_IWUSR)) {
+              throw OptException("Missing write permissions for"
```

Comment:
I tried `/Daten/Git/jlab/gapc>./out -O /etc/klkdjff AUCGUGCA` and got the message:
```
Checkpointing routine has been integrated. A new checkpoint will be created every 0 days, 1 hours, 0 minutes and 0 seconds.
The checkpoints will be saved at "/etc".

Answer: 
1
```
Which surprises me as I assumed a normal user cannot write to `/etc`?!

Reply:
Yeah that was bad programming on my part. I fixed that.